src/Hilke.KineticConvolution/Helpers/VectorCalculator.cs

```diff
@@ -0,0 +1,249 @@
+﻿using System;
+using System.Linq;
+
+namespace Hilke.KineticConvolution.Helpers
+{
+    public class VectorCalculator<TAlgebraicNumber>
+    {
+        private readonly IAlgebraicNumberCalculator<TAlgebraicNumber> _calculator;
+        private readonly ConvolutionFactory<TAlgebraicNumber> _factory;
+
+        public VectorCalculator(IAlgebraicNumberCalculator<TAlgebraicNumber> calculator)
+        {
+            _calculator = calculator ?? throw new ArgumentNullException(nameof(calculator));
+            _factory = new ConvolutionFactory<TAlgebraicNumber>(_calculator);
+        }
+
+        public TAlgebraicNumber GetDot(Vector<TAlgebraicNumber> left, Vector<TAlgebraicNumber> right)
+        {
+            if (left is null)
+            {
+                throw new ArgumentNullException(nameof(left));
+            }
+
+            if (right is null)
+            {
+                throw new ArgumentNullException(nameof(right));
+            }
+
+            if (left.Dimension != right.Dimension)
+            {
+                throw new ArgumentException("The vectors must have the same dimension.");
+            }
+
+            return left.Coordinates.Zip(
+                           right.Coordinates,
+                           (leftCoordinate, rightCoordinate) => _calculator.Multiply(leftCoordinate, rightCoordinate))
+                       .Aggregate((sum, next) => _calculator.Add(sum, next));
```

Comment:
This can be simplified:
```suggestion
            return left.Coordinates
                .Zip(right.Coordinates, _calculator.Multiply)
                .Aggregate(_calculator.Add);
```